internal/usecase/payment_usecase.go

```diff
@@ -0,0 +1,148 @@
+package usecase
+
+import (
+	"context"
+
+	"github.com/tubagusmf/payment-service-gb1/internal/model"
+
+	"github.com/go-playground/validator"
+	"github.com/labstack/gommon/log"
+	"github.com/sirupsen/logrus"
+)
+
+type PaymentUsecase struct {
+	paymentRepo model.IPaymentRepository
+	// workerClient *worker.AsynqClient
+}
+
+var v = validator.New()
+
+func NewPaymentUsecase(
+	paymentRepo model.IPaymentRepository,
+	// workerClient *worker.AsynqClient,
+) model.IPaymentUsecase {
+	return &PaymentUsecase{
+		paymentRepo: paymentRepo,
+		// workerClient: workerClient,
+	}
+}
+
+func (p *PaymentUsecase) FindAll(ctx context.Context, filter model.PaymentFilter) ([]*model.Payment, error) {
+	log := logrus.WithFields(logrus.Fields{
+		"ctx":    ctx,
+		"limit":  filter.Limit,
+		"offset": filter.Offset,
+	})
+
+	payments, err := p.paymentRepo.FindAll(ctx, filter)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	return payments, nil
+}
+
+func (p *PaymentUsecase) FindById(ctx context.Context, id int64) (*model.Payment, error) {
+	log := logrus.WithFields(logrus.Fields{
+		"ctx": ctx,
+		"id":  id,
+	})
+
+	payment, err := p.paymentRepo.FindById(ctx, id)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	return payment, nil
+}
+
+func (p *PaymentUsecase) Create(ctx context.Context, in model.CreatePaymentInput) error {
+	log := logrus.WithFields(logrus.Fields{
+		"ctx":       ctx,
+		"name":      in.Name,
+		"bank_code": in.BankCode,
+	})
+
+	err := p.validateCreatePaymentInput(ctx, in)
```

Comment:
terkait ini, validate input seharusnya jgn di layer usecase, di handler. usecase focus sama logic aja mas @tubagusmf

Reply:
menurutku gapapa soalnya validasi bisa berhubungan dengan logic bisnis juga misal nama payment gaboleh pake symbol dll

https://stackoverflow.com/questions/57603422/clean-architecture-where-to-put-input-validation-logic